### ISSUE TITLE
https://github.com/bndtools/workspace is out of date

### ISSUE THREAD

**Issue by BryanHunt** [bndtools/bnd#1594]
This can be reproduced from: https://github.com/BryanHunt/bnd-test

I have a workspace that contains only one bundle that is an integration test.  If you start with an empty ~/.bnd directory, the project will fail to resolve bundles on the check task with errors like:

```
java.io.IOException: Could not open URL, [404] Not Found for 'http://www.mirrorservice.org/sites/ftp.apache.org//felix/org.apache.felix.configadmin-1.8.8.jar'
	at aQute.bnd.jpm.StoredRevisionCache.getConnection(StoredRevisionCache.java:201)
	at aQute.bnd.jpm.StoredRevisionCache.doDownload(StoredRevisionCache.java:144)
	at aQute.bnd.jpm.StoredRevisionCache.download(StoredRevisionCache.java:104)
	at aQute.bnd.jpm.Repository$1.run(Repository.java:343)
java.io.IOException: Could not open URL, [404] Not Found for 'http://www.mirrorservice.org/sites/ftp.apache.org//felix/org.apache.felix.webconsole-4.2.14-all.jar'
	at aQute.bnd.jpm.StoredRevisionCache.getConnection(StoredRevisionCache.java:201)
	at aQute.bnd.jpm.StoredRevisionCache.doDownload(StoredRevisionCache.java:144)
	at aQute.bnd.jpm.StoredRevisionCache.download(StoredRevisionCache.java:104)
	at aQute.bnd.jpm.Repository$1.run(Repository.java:343)
java.io.IOException: Could not open URL, [404] Not Found for 'http://www.mirrorservice.org/sites/ftp.apache.org//felix/org.apache.felix.http.jetty-3.1.6.jar'
	at aQute.bnd.jpm.StoredRevisionCache.getConnection(StoredRevisionCache.java:201)
	at aQute.bnd.jpm.StoredRevisionCache.doDownload(StoredRevisionCache.java:144)
	at aQute.bnd.jpm.StoredRevisionCache.download(StoredRevisionCache.java:104)
	at aQute.bnd.jpm.Repository$1.run(Repository.java:343)
```

This causes Jenkins pipeline builds to fail.

**Comment by bjhargrave** [comment]
As I noted in https://github.com/BryanHunt/bnd-test/pull/1, I don't see these errors.

I think you may have connectivity issue from your CI server to those URLs. See if your CI server can `curl` or `wget` those URLs.

**Comment by BryanHunt** [comment]
If you open a browser on http://www.mirrorservice.org/sites/ftp.apache.org//felix you will see that the bundle versions used above do not exist.  It seems that the changes to my project are to use "${build}/cache/.bnd/shacache" instead of ~/.bnd/shacache and +bnd_plugin=biz.aQute.bnd:biz.aQute.bnd.gradle:3.2.0 instead of -bnd_plugin=biz.aQute.bnd:biz.aQute.bnd.gradle:3.0.0.  The bndtools/worspace is how I got those original values.  Does it need to be updated?

**Comment by bjhargrave** [comment]
> "${build}/cache/.bnd/shacache" instead of ~/.bnd/shacache and

Is just a change I made to avoid having to delete my ~/.bnd/shacache to test your problem. So that does not need to go in any template.



> +bnd_plugin=biz.aQute.bnd:biz.aQute.bnd.gradle:3.2.0 instead of -bnd_plugin=biz.aQute.bnd:biz.aQute.bnd.gradle:3.0.0

The template should use the version of bndtools you are running for the version of the bnd gradle plugin when creating a new workspace.

**Comment by bjhargrave** [comment]
> If you open a browser on http://www.mirrorservice.org/sites/ftp.apache.org//felix you will see that the bundle versions used above do not exist.

You have those URLs in your `cnf/central.json` file: <https://github.com/BryanHunt/bnd-test/blob/master/cnf/central.json#L3>. If they are invalid URLs, then you should have them in that file.

**Comment by BryanHunt** [comment]
I did not put them in that file, bndtools did.

**Comment by BryanHunt** [comment]
Changing gradle.properties to use biz.aQute.bnd:biz.aQute.bnd.gradle:3.2.0 fixes the problem.

If you create a new workspace and create a new cnf, biz.aQute.bnd:biz.aQute.bnd.gradle:3.0.0 is the default.

**Comment by bjhargrave** [comment]
> I did not put them in that file, bndtools did.

I am not sure where your jpm config came from.



> If you create a new workspace and create a new cnf, biz.aQute.bnd:biz.aQute.bnd.gradle:3.0.0 is the default.

What version of bndtools are you using?

**Comment by BryanHunt** [comment]
![eclipse_installation_details_and_nvdocumentservice_-_bndtools_-_compare__cnf_central_json_current_and_1686deb_-_eclipse](https://cloud.githubusercontent.com/assets/504299/17679336/a2c5b276-6300-11e6-9e7e-033748246cb1.jpg)

**Comment by bjhargrave** [comment]
This seems to be causes by https://github.com/bndtools/workspace which is offered by Bndtools as a workspace template. But it seems to be very out of date and contains the bad URLs. @njbartlett created it. I am not sure if he is maintaining it. But it seems like it should not be used in its current state.

**Comment by bjhargrave** [comment]
![untitled](https://cloud.githubusercontent.com/assets/277682/17679705/9c24649e-630a-11e6-8cb4-d6d92ac086c8.jpg)

**Comment by seanbright** [comment]
See https://github.com/bndtools/workspace/pull/1

**Comment by seanbright** [comment]
@BryanHunt Can you test against bndtools/workspace, branch 3.3 instead of the OSGi_R6 tag?

**Comment by bjhargrave** [comment]
> @BryanHunt Can you test against bndtools/workspace, branch 3.3 instead of the OSGi_R6 tag?

The 3.3 branch gralde build wont work know because it is set up to use the as yet unreleased 3.3 bnd gradle plugin. If you don't care about the gradle build, then 3.3 branch is fine. Otherwise test the master branch which is set up to use the released 3.2 gradle plugin.
